src/components/20-molecules/image-upload/index.js

```diff
@@ -0,0 +1,433 @@
+                  />
+                `}
+            <div class="m-image-upload__icon-layer">
+              <span class="m-image-upload__icon-error"
+                >${isfaultyFile ? ClearIcon : ''}</span
+              >
+              <span class="m-image-upload__icon-delete"
+                >${DeleteForeverIcon}</span
+              >
+            </div>
+          </div>
+          ${isfaultyFile
+            ? html`
+                <figcaption
+                  class="m-image-upload__img-caption js-image-upload__img-caption m-image-upload__img-caption-error"
+                  title="${fileTooBigStatusText}"
+                  data-status="${deleteStatusText}"
+                >
+                  <span
+                    class="m-image-upload__filename js-image-upload__filename"
+                    >${fileTooBigStatusText}</span
+                  >
+                </figcaption>
+              `
+            : html`
+                <figcaption
+                  class="m-image-upload__img-caption js-image-upload__img-caption"
+                  title="${file.name}"
+                  data-status="${deleteStatusText}"
+                >
+                  <span
+                    class="m-image-upload__filename js-image-upload__filename"
+                    >${file.name}</span
+                  >
+                </figcaption>
+              `}
+        </figure>
+      `;
+    });
+  }
+
+  generateAddMoreInputFile() {
+    const { addStatusText } = this;
+    return html`
+      <figure
+        class="m-image-upload__img-figure js-image-upload__img-figure m-image-upload__add-more js-image-upload__add-more"
+      >
+        <div
+          class="m-image-upload__icon-wrapper"
+          @click=${this.handleAddMoreInputClick}
+        >
+          <div class="m-image-upload__icon-layer">
+            ${AddIcon}
+          </div>
+        </div>
+        <figcaption
+          class="m-image-upload__img-caption js-image-upload__img-caption"
+          title="${addStatusText}"
+        >
+          ${addStatusText}
+        </figcaption>
+      </figure>
+    `;
+  }
+
+  render() {
+    const imageOverviewClasses = {
+      'm-image-upload__dropzone': true,
+      'js-image-upload__dropzone': true,
+      'm-image-upload__dropzone-file-overview': this.showImageOverview,
+      'js-image-upload__dropzone-file-overview': this.showImageOverview,
+    };
+    const errorMessageWrapperClasses = {
+      'm-image-upload__error-wrapper': true,
+      'js-image-upload__error-wrapper': true,
+    };
+
+    /* displaying files with errors (e.g. too big) after valid ones */
+    this.allFiles = this.files.concat(this.faultyFiles);
+    const fileOverview = this.fileOverviewMapping(this.allFiles);
+    this.addMoreInputFile = this.generateAddMoreInputFile();
+
+    return html`
+      <article class="m-image-upload">
+        <h1><slot></slot></h1>
+        <section
+          @dragover="${this.handleImageUploadDropZoneDragover}"
```

Comment:
Yes that's because I only use it ones. In my opinion using destructuring on every class level variable just gets way too messy. But that's personal preference. I think this is a good subject to discuss in the team. Are you in?